lib/siwapp/recurring_invoices.ex

```diff
@@ -49,20 +55,39 @@ defmodule Siwapp.RecurringInvoices do
     Repo.delete(recurring_invoice)
   end
 
-  # Expect will be added when function is finished
+  @doc """
+  Generates invoices associated to recurring_invoice if this is enabled
+  """
+  @spec generate_invoices(pos_integer()) :: [{:ok, Invoice.t()}] | nil
   def generate_invoices(id) do
-    Repo.get!(RecurringInvoice, id)
+    rec_inv = Repo.get!(RecurringInvoice, id)
+
+    if invoices_to_generate(id) > 0 do
+      for _i <- 1..invoices_to_generate(id),
+          do: Invoices.create(build_invoice_attrs(rec_inv))
```

Comment:
Estás invocando 2 veces el invoices_to_generate y además todo este if+for se podría hacer en un simple Enum.each